### PR TITLE
[feature](function) add ip function ipv4numtostring (alias inet_ntoa)

### PR DESCRIPTION
**Pull request by Yulei-Yang**
## Proposed changes

Issue Number: #20584 

image

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Yulei-Yang** [comment]
run buildall

**Comment by Yulei-Yang** [comment]
run beut

**Comment by Yulei-Yang** [comment]
run buildall

**Comment by Yulei-Yang** [comment]
run p0

**Comment by Yulei-Yang** [comment]
run buildall

**Comment by Yulei-Yang** [comment]
run buildall

**Comment by Yulei-Yang** [comment]
run buildall

**Comment by Yulei-Yang** [comment]
run p0

**Comment by Yulei-Yang** [comment]
run buildall

**Review by nextdreamblue — DISMISSED** [review]

**Comment by Yulei-Yang** [comment]
run buildall

**Comment by Yulei-Yang** [comment]
run clickbench

**Comment by Yulei-Yang** [comment]
run beut

**Comment by Yulei-Yang** [comment]
run buildall

**Review by nextdreamblue — APPROVED** [review]

**Comment by nextdreamblue** [comment]
LGTM

File: regression-test/suites/query_p0/sql_functions/ip_functions/test_ip_functions.groovy [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+// specific language governing permissions and limitations
+// under the License.
+suite("test_ip_functions") {
+    sql "set batch_size = 4096;"
+
+    qt_sql "SELECT ipv4numtostring(-1);"
+    qt_sql "SELECT ipv4numtostring(2130706433);"
+    qt_sql "SELECT ipv4numtostring(4294967298);"
+    qt_sql "SELECT inet_ntoa(3232235521);"
+
```
Comment (zclllyybb) [inline regression-test/suites/query_p0/sql_functions/ip_functions/test_ip_functions.groovy:24]: need a testcase which argument is a table column

**Review by zclllyybb — CHANGES_REQUESTED** [review]

**Review by yuxuan-luo — APPROVED** [review]
LGTM

File: regression-test/suites/query_p0/sql_functions/ip_functions/test_ip_functions.groovy (hunk 1 above)
Comment (Yulei-Yang) [reply regression-test/suites/query_p0/sql_functions/ip_functions/test_ip_functions.groovy:24]: will do it in next pr, pipeline of this pr cost too much time

**Comment by wangchong2023** [comment]
Thanks very much for your help.There are some IPV4/IPv6 functions(not only translation between digital and string). Is there plan to implement them?

**Pull request merged**